eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -195,34 +195,51 @@ def get_sources_for_taxonomy_assembly(private_config_xml_file, profile, release_
             return result[0][0]
 
 
-def find_if_any_ss_has_rs_for_tax_and_asm(mongo_source, coll, tax, asm):
-    search_query = {'tax': tax, 'seq': asm, 'rs': {'$exists': True}}
-    collection = mongo_source.mongo_handle[mongo_source.db_name][coll]
-    result = collection.find_one(search_query)
-    if result:
-        logger.info(f'Found SS with RS for Taxonomy {tax} and Assembly {asm} : SS {result}')
+def determine_should_be_released_for_coll(mongo_source, tax, asm, ss_coll, rs_coll, ss_query, rs_query):
+    logger.info(f"Looking for SS with RS for Taxonomy {tax} and Assembly {asm} in collection {ss_coll}")
+    collection = mongo_source.mongo_handle[mongo_source.db_name][ss_coll]
+    ss_with_rs = collection.find_one(ss_query)
+    if ss_with_rs:
+        logger.info(f'Found SS with RS for Taxonomy {tax} and Assembly {asm} in collection {ss_coll}, SS: {ss_with_rs}')
         return True
     else:
-        logger.warning(f'No SS with RS found for Taxonomy {tax} and Assembly {asm} in collection {coll}')
+        logger.warning(f'No SS with RS found for Taxonomy {tax} and Assembly {asm} in collection {ss_coll}')
+
+    logger.info(f"Looking for RS with Taxonomy {tax} and Assembly {asm} in collection {rs_coll}")
+    collection = mongo_source.mongo_handle[mongo_source.db_name][rs_coll]
+    rs_with_tax_asm = collection.find_one(rs_query)
+    if rs_with_tax_asm:
+        logger.info(f'Found RS with Taxonomy {tax} and Assembly {asm} in collection {rs_coll}, RS: {rs_with_tax_asm}')
+        return True
```

Comment:
I am a bit confused as to why we do this? Even if there is a special reason, it should be documented here...

Reply:
There are cases where SVE/dbsnpSVE has a variant but the corresponding CVE/dbsnpCVE does not have any. 
This was added for the opposite case where SS might not have a variant in a collection but there is a variant corresponding CVE/dbsnpCVE collection because of operations like split..